src/widgets/training-program/ui/training-program.test.tsx

```diff
@@ -1,50 +1,13 @@
 import { screen } from '@testing-library/react';
 import { beforeEach } from 'vitest';
-import { ROUTES } from '@/core/const';
 import { Course } from '@/entities/course';
-import { MOCKED_IMAGE_PATH } from '@/shared/__tests__/constants';
+import { mockedCourses } from '@/shared/__tests__/constants';
 import { renderWithRouter } from '@/shared/__tests__/utils';
-import { COURSE_ALIASES, TO_BE_DETERMINED } from '@/shared/constants';
 import { TrainingProgram } from '@/widgets/training-program';
 import { COURSE_TITLES } from 'data';
 
-const mockedCourseAngular: Course = {
-  id: '1',
-  title: COURSE_TITLES.ANGULAR,
-  alias: COURSE_ALIASES.ANGULAR,
-  startDate: '16 Oct, 2023',
-  registrationEndDate: '16 Oct, 2024',
-  iconSmall: MOCKED_IMAGE_PATH,
-  iconSrc: MOCKED_IMAGE_PATH,
-  secondaryIcon: MOCKED_IMAGE_PATH,
-  language: ['en'],
-  mode: 'online',
-  detailsUrl: `/${ROUTES.COURSES}/${ROUTES.ANGULAR}`,
-  enroll: 'https://wearecommunity.io/events/rs-angular-2023q4',
-  backgroundStyle: {
-    backgroundColor: '#F4F1FA',
-    accentColor: '#F4AFA7',
-  },
-};
-
-const mockedCourseAws: Course = {
-  id: '8',
-  language: ['en'],
-  iconSmall: MOCKED_IMAGE_PATH,
-  iconSrc: MOCKED_IMAGE_PATH,
-  secondaryIcon: MOCKED_IMAGE_PATH,
-  mode: 'online',
-  startDate: '',
-  registrationEndDate: TO_BE_DETERMINED,
-  title: COURSE_TITLES.AWS_CLOUD_DEVELOPER,
-  alias: COURSE_ALIASES.AWS_CLOUD_DEVELOPER,
-  detailsUrl: `/${ROUTES.COURSES}/${ROUTES.AWS_DEVELOPER}`,
-  enroll: 'https://wearecommunity.io/events/aws-cloud-dev-rs2023q4',
-  backgroundStyle: {
-    backgroundColor: '#F4F1FA',
-    accentColor: '#F4AFA7',
-  },
-};
+const mockedCourseAngular: Course = mockedCourses[4];
+const mockedCourseAws: Course = mockedCourses[5];
 
 describe('TrainingProgram', () => {
   describe('with "angular" props', () => {
```

Comment:
The test uses `getByText` to find the title, but we could have used `data-testid=“widget-title”.` And for paragraphs use `data-testid=“paragraph”.` Wouldn't it be better to switch to a `data-testid` search?

Reply:
done for titles. There is a problem with paragraphs because there is more than 1 paragraph, but `testid` only select the first one, so in order to determine the right one is adding extra complexity